File: .agent/skills/rb-ghl-automation/scripts/create_contact.js
const { makeRequest } = require('./common_ghl');

async function createContact(firstName, lastName, email, phone, tags) {
    const endpoint = "/contacts/";

    const payload = {
        firstName: firstName,
        lastName: lastName,
        email: email,
        phone: phone,
        tags: tags ? tags.split(',') : []
    };

    // Remove undefined/null values
    Object.keys(payload).forEach(key => (payload[key] === undefined || payload[key] === null) && delete payload[key]);

    const resp = await makeRequest("POST", endpoint, {}, payload);

    if (resp && resp.contact) {
        console.log(`✅ Contact Created/Updated: ${resp.contact.id}`);
        console.log(`Name: ${resp.contact.firstName} ${resp.contact.lastName}`);
    } else {
        console.log("❌ Failed to create contact.");
    }
}

// Simple CLI argument parsing
const args = process.argv.slice(2);
const helpIndex = args.indexOf('--help');

if (helpIndex !== -1 || args.length === 0) {
    console.log("Usage: node create_contact.js --email <email> [--first <first>] [--last <last>] [--phone <phone>] [--tags <tag1,tag2>]");
    process.exit(0);
}

function getArg(name) {
    const index = args.indexOf(name);
    return index !== -1 ? args[index + 1] : null;
}

const email = getArg('--email');
const firstName = getArg('--first');
const lastName = getArg('--last');
const phone = getArg('--phone');
const tags = getArg('--tags');

if (!email) {
    console.error("Error: --email is required");
    process.exit(1);
}

createContact(firstName, lastName, email, phone, tags);

File: .agent/skills/rb-ghl-automation/scripts/create_contact.py
import argparse
from common_ghl import make_request

def create_contact(first_name, last_name, email, phone, tags):
    endpoint = "/contacts/"
    
    payload = {
        "firstName": first_name,
        "lastName": last_name,
        "email": email,
        "phone": phone,
        "tags": tags if tags else []
    }
    
    # Remove None values
    payload = {k: v for k, v in payload.items() if v is not None}
    
    resp = make_request("POST", endpoint, data=payload)
    
    if resp and "contact" in resp:
        c = resp["contact"]
        print(f"✅ Contact Created/Updated: {c.get('id')}")
        print(f"Name: {c.get('firstName')} {c.get('lastName')}")
    else:
        print("❌ Failed to create contact.")

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Create/Update a GHL Contact")
    parser.add_argument("--first", help="First Name")
    parser.add_argument("--last", help="Last Name")
    parser.add_argument("--email", required=True, help="Email Address")
    parser.add_argument("--phone", help="Phone Number")
    parser.add_argument("--tags", nargs="+", help="List of tags (e.g. RB-Lead RB-Booked)")
    
    args = parser.parse_args()
    
    create_contact(args.first, args.last, args.email, args.phone, args.tags)
